argilla/src/argilla/datasets/_export/_disk.py

```diff
@@ -98,20 +98,29 @@ def from_disk(
             workspace_id = client.workspaces.default.id
         dataset_model.workspace_id = workspace_id
 
-        # Get a relevant and unique name for the incoming dataset.
-        if name:
-            logging.warning(f"Changing dataset name from {dataset_model.name} to {name}")
+        if name and (name != dataset_model.name):
+            logging.info(f"Changing dataset name from {dataset_model.name} to {name}")
             dataset_model.name = name
-        elif client.api.datasets.name_exists(name=dataset_model.name, workspace_id=workspace_id):
-            logging.warning(f"Loaded dataset name {dataset_model.name} already exists. Changing to unique UUID.")
-            dataset_model.name = f"{dataset_model.name}_{uuid4()}"
 
+        if client.api.datasets.name_exists(name=dataset_model.name, workspace_id=workspace_id):
+            warnings.warn(
+                f"Loaded dataset name {dataset_model.name} already exists in the workspace so using it. To create a new dataset, provide a unique name to the `name` parameter."
```

Comment:
Maybe we can add info about the workspace here
```suggestion
                f"Loaded dataset name {dataset_model.name} already exists in the workspace {workspace.name} so using it. To create a new dataset, provide a unique name to the `name` parameter."
```